modules/ListView.js

```diff
@@ -366,31 +405,74 @@
       return model ? this.getChildView(this.__modelToViewMap[model[this.__modelId]]) : undefined;
     },
 
+    /**
+     * @returns {Boolean} returns true if there exists any generated item views
+     * @method hasItemViews
+     */
+    hasItemViews: function() {
+      return !_.isEmpty(this.getItemViews());
+    },
+
+    /**
+     * @returns {Array of views} Returns unordered list of views generated by this list view
+     * @method getItemViews
+     */
+    getItemViews: function() {
+      var view = this;
+      var orderedViewIds = _.map(this.__orderedModelIdList, function(modelId) { return view.__modelToViewMap[modelId]; });
+      return _.map(orderedViewIds, this.getChildView, this);
+    },
+
     /************** Private methods **************/
 
     /**
-     * Creates a new child view if there doesn't exist one for a model
-     * @method __createChildViews
+     * Creates a new item view for a model if there doesn't exist one
+     * @method __createItemViews
      * @private
      */
-    __createChildViews: function() {
-      var newChildViews = [];
-      _.each(this.modelsToRender(), function(model) {
+    __createItemViews: function() {
+      var newItemViews = [];
+      _.each(this.modelsToRender(), function(model, indexOfModel) {
         var childView = this.getChildViewFromModel(model);
         if (!childView) {
-          newChildViews.push(this.__createChildView(model));
+          newItemViews.push({ view: this.__createItemView(model, true), indexOfModel: indexOfModel });
         }
       }, this);
-      return newChildViews;
+      this.__updateOrderedModelIdList();
+      return newItemViews;
     },
 
     /**
-     * Removes a child view's that have models that are no longer tracked by modelsToRender
-     * @method __removeStaleChildViews
+     * Creates an item view and stores a reference to it
+     * @method __createItemView
      * @private
+     * @param model {Backbone Model} the model to create the view from
+     * @param [noUpdateToIdList=false] if true, the internal order of model ids are not updated
+     * @return {Backbone View} the new item view
      */
-    __removeStaleChildViews: function() {
-      var removedChildViews = [];
+    __createItemView: function(model, noUpdateToIdList) {
+      var itemView,
+        ItemViewClass = this.childView;
+      if (!_.isFunction(this.childView.extend)) {
+        ItemViewClass = this.childView(model);
+      }
+      itemView = new ItemViewClass(this.__generateChildArgs(model));
+      this.registerTrackedView(itemView, { shared: false });
+      this.__modelToViewMap[model[this.__modelId]] = itemView.cid;
```

Comment:
shouldn't this be model.get(this.__modelId) ?  Instead of model[this.__modelId].

Oh I keep forgetting this is the name of the model id, like cid or something else?  What else would you reference again that is useful?  id?  Is there any other option?